scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,356 @@
+import numpy as np
+
+
+def _validate_1d(a, name):
+    if np.ndim(a) != 1:
+        raise ValueError(f'{name} must be a one-dimensional sequence')
+    if np.isnan(a).any():
+        raise ValueError(f'{name} must not contain nan.')
+    if np.isinf(a).any():
+        raise ValueError(f'{name} must contain only finite values')
+
+
+def _validate_intervals(intervals):
+    intervals = np.asarray(intervals)
+    if intervals.shape == (0,):
+        # The input was a sequence with length 0.
+        intervals = intervals.reshape((0, 2))
+    if intervals.ndim != 2 or intervals.shape[-1] != 2:
+        raise ValueError('intervals must be a two-dimensional array'
+                         ' with shape (m, 2)')
```

Comment:
(On behalf of the user) What is `m`?

Reply:
Also, I think this is untested. I'll check codecov.

Reply:
I made the error more informative in https://github.com/scipy/scipy/pull/13699/commits/f4b0b7a77ad970c80b55158fb02b90d8de6838c4